tests/client/test_upload.py

```diff
@@ -29,7 +29,9 @@ def test_upload(dryrun, caplog):
 
     # Expected calls to Bucket methods objects.filters and upload_file
     objects_calls = [mock.call(Prefix=item.key) for item in items]
-    upload_calls = [mock.call(item.path, item.key) for item in items]
+    upload_calls = [
+        mock.call(item.path, item.key, ExtraArgs=None) for item in items
```

Comment:
It seems like there ought to be at least one test with a non-None ExtraArgs during upload - otherwise wouldn't all the tests still pass even if the code were broken such that headers were ignored during upload?